src/main/java/com/cinema/config/SecurityConfig.java

```diff
@@ -1,30 +1,44 @@
 package com.cinema.config;
 
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.annotation.Bean;
+import org.springframework.http.HttpMethod;
 import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
 import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
-import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.core.userdetails.UserDetailsService;
 import org.springframework.security.crypto.password.PasswordEncoder;
 
 @EnableWebSecurity
 public class SecurityConfig extends WebSecurityConfigurerAdapter {
+    private final UserDetailsService userDetailsService;
+    private final PasswordEncoder encoder;
+
+    public SecurityConfig(UserDetailsService userDetailsService, PasswordEncoder encoder) {
+        this.userDetailsService = userDetailsService;
+        this.encoder = encoder;
+    }
 
     @Autowired
     public void configureGlobal(AuthenticationManagerBuilder auth) throws Exception {
         auth
-                .inMemoryAuthentication()
-                .passwordEncoder(getEncoder())
-                .withUser("admin@gmail.com")
-                .password(getEncoder().encode("1234"))
-                .roles("ADMIN");
+                .userDetailsService(userDetailsService)
+                .passwordEncoder(encoder);
     }
 
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
+                .and()
+                .authorizeRequests()
+                .antMatchers(HttpMethod.POST, "/movies/**").hasRole("ADMIN")
+                .antMatchers(HttpMethod.GET, "/movies/**").permitAll()
+                .antMatchers(HttpMethod.POST, "/cinema-halls/**").hasRole("ADMIN")
+                .antMatchers(HttpMethod.GET, "/cinema-halls/**").permitAll()
+                .antMatchers(HttpMethod.POST, "/movie-sessions/**").hasRole("ADMIN")
+                .antMatchers(HttpMethod.GET, "/movie-sessions/**").permitAll()
+                .antMatchers("/shopping-carts/**").hasRole("CLIENT")
+                .antMatchers("/orders/").hasRole("CLIENT")
```

Comment:
you can merge matchers if method is the same and role is the same. Just divide URLs by comma